### PR TITLE
Send and display headers as title case

### LINKED ISSUE #166: xh lowercases all headers
## What I expect
Xh copies case of user-specified headers.

## What actually happens
Xh treats all headers as case-insensitive and lowercases them.

## Why is this a problem?
1. It's a difference from httpie.
2. It has a negative net effect when interacting with non-compliant servers.

To explain 2...

Some services treat headers as case-sensitive (e.g with `X-*` headers, especially for auth).
The HTTP specification does specify that headers are case-insensitive, however, consider the following matrix.
If headers are treated as case insensitive, this will break servers that (incorrectly) treat them as case-sensitive, while not having a negative effect on compliant servers.
If headers are treated as case sensitive, this will work with the non-compliant servers, while not having a negative effect on compliant servers.

It would also be ok to auto-capitalize headers (e.g `foO-baR-baZ` could become `Foo-Bar-Baz`), since that's what standard industry practice anyway, but this is not what HTTPie does.

## How to test compliance
`xh --print=H localhost My-OdDlY-Capitalized-Header:Here`
compare to
`http --print=H localhost My-OdDlY-Capitalized-Header:Here`

### PR DESCRIPTION
**Pull request by blyxxyz**
Resolves #166.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,6 +162,7 @@ xh -d httpbin.org/json -o res.json
 
 - Not all of HTTPie's features are implemented. ([#4](https://github.com/ducaale/xh/issues/4))
 - HTTP/2 cannot be disabled. ([#68](https://github.com/ducaale/xh/issues/68))
+- Headers are not case-sensitive.
```
Comment (ducaale) [inline README.md:165]: It might be good to clarify that header values are case-sensitive

File: src/printer.rs [hunk 2]
```diff
@@ -564,4 +598,40 @@ mod tests {
...
+        );
+
+        assert_eq!(
+            p.headers_to_string(&headers, reqwest::Version::HTTP_2),
+            "ab-cd: 0
+-cd: 0
+-: 0
+ab-%c: 0
+a-b--c: 0"
+        );
```
Comment (ducaale) [inline src/printer.rs:635]: We can use the `indoc!` macro here just like we do in `tests/cli.rs`.

**Review by ducaale — COMMENTED** [review]
Looks good. I have two minor comments

**Review by ducaale — APPROVED** [review]
LGTM

**Comment by rburgst** [comment]
IMHO sending headers as title case by default is somewhat unexpected (and certainly different from how `httpie` does it. 
Note that by RFC headers should be case insensitive therefore, on the server you should usually only expect lowercase headers. 
Now if I send a request
```
xh http://localhost:3000 foo:bar
```

`xh`  will actually send `Foo: bar` to the server which is unexpected IMHO.

**Pull request merged**